source/PointCloud.cc

```diff
@@ -0,0 +1,115 @@
+/* Copyright (c) 2023, the adamantine authors.
+ *
+ * This file is subject to the Modified BSD License and may not be distributed
+ * without copyright and license information. Please refer to the file LICENSE
+ * for the text and further information on this license.
+ */
+
+#include <PointCloud.hh>
+#include <instantiation.hh>
+
+#include <boost/filesystem.hpp>
+
+#include <fstream>
+#include <regex>
+
+namespace adamantine
+{
+template <int dim>
+PointCloud<dim>::PointCloud(
+    boost::property_tree::ptree const &experiment_database)
+{
+  // Format of the file names: the format is pretty arbitrary, #frame and
+  // #camera are replaced by the frame and the camera number.
+  // PropertyTreeInput experiment.file
+  _data_filename = experiment_database.get<std::string>("file");
+  // PropertyTreeInput experiment.first_frame
+  _next_frame = experiment_database.get("first_frame", 0);
+  // PropertyTreeInput experiment.first_camera_id
+  _first_camera_id = experiment_database.get<unsigned int>("first_camera_id");
+  // PropertyTreeInput experiment.last_camera_id
+  _last_camera_id = experiment_database.get<int>("last_camera_id");
+}
+
+template <int dim>
+unsigned int PointCloud<dim>::read_next_frame()
+{
+  _points_values_current_frame.points.clear();
+  _points_values_current_frame.values.clear();
+  for (unsigned int camera_id = _first_camera_id;
+       camera_id < _last_camera_id + 1; ++camera_id)
+  {
+    // Use regex to get the next file to read
+    std::regex camera_regex("#camera");
+    std::regex frame_regex("#frame");
+    auto filename =
+        std::regex_replace((std::regex_replace(_data_filename, camera_regex,
+                                               std::to_string(camera_id))),
+                           frame_regex, std::to_string(_next_frame));
+    unsigned int counter = 1;
+    while (!boost::filesystem::exists(filename))
+    {
+      // Spin loop waiting for the file to appear
+      if (counter == 0)
+        std::cout << "Waiting for the next frame" << std::endl;
+      ++counter;
+    }
```

Comment:
Should the counter be initialized to 0 here instead of 1? I assume the purpose of this is to print the message once and then quietly increment until the file exists?

Reply:
I don't want to see this message. I've put it there because it's a potentially infinite loop. If this message appears, I can check that the frame really does not exist. It's more of a debugging tool.

Reply:
Just to make sure I'm understanding -- the idea is that the message appears only if the file doesn't exist for long enough for `counter` to overflow and make it back to zero?

Reply:
Yes, exactly

Reply:
Would you mind changing the comment to read:
```suggestion
    unsigned int counter = 1;
    while (!boost::filesystem::exists(filename))
    {
      // Spin loop waiting for the file to appear (message printed if counter overflows)
      if (counter == 0)
        std::cout << "Waiting for the next frame" << std::endl;
      ++counter;
    }
```